src/Atmos/Model/AtmosModel.jl

```diff
@@ -474,7 +474,20 @@ end
 @inline function wavespeed(m::AtmosModel, nM, state::Vars, aux::Vars, t::Real)
     ρinv = 1 / state.ρ
     u = ρinv * state.ρu
-    return abs(dot(nM, u)) + soundspeed(m, m.moisture, state, aux)
+    uN = abs(dot(nM, u))
+    ss = soundspeed(m, m.moisture, state, aux)
+
+    FT = typeof(state.ρ)
+    ws = -@MVector(zeros(FT, number_state_conservative(m, FT)))
+    vars_ws = Vars{vars_state_conservative(m, FT)}(ws)
+
+    ws .= uN
```

Comment:
Not sure if it makes any difference, but you could do `ws = fill(uN, MVector{number_state_conservative(m, FT), FT})`
to create a vector filled with the value `uN`.

Reply:
Looks good.  I will update it as I try to debug.